app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,610 @@
+                return;
+            }
+
+            setLoading(true);
+
+            if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+                const {data: userData} = await searchProfiles(
+                    serverUrl, text.toLowerCase(),
+                    {team_id: currentTeamId, allow_inactive: true});
+
+                if (userData) {
+                    setSearchResults(userData);
+                }
+            } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+                const isSearch = true;
+                const {channels: receivedChannels} = await searchChannels(
+                    serverUrl, text, currentTeamId, isSearch);
+
+                if (receivedChannels) {
+                    setSearchResults(receivedChannels);
+                }
+            } else if (dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+                await searchDynamicOptions(text);
+            }
+
+            setLoading(false);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [dataSource]);
+
+    const getMultiselectData = useCallback((multiselectSelectedElems: MultiselectSelectedMap): Selection => {
+        let myItems;
+        let multiselectItems: Selection = [];
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                myItems = multiselectSelectedElems as Dictionary<UserProfile>;
+                multiselectItems = multiselectItems as UserProfile[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                myItems = multiselectSelectedElems as Dictionary<Channel>;
+                multiselectItems = multiselectItems as Channel[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            default:
+                myItems = multiselectSelectedElems as Dictionary<DialogOption>;
+                multiselectItems = multiselectItems as DialogOption[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+        }
+    }, [multiselectSelected, dataSource]);
+
+    // Effects
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleMultiselectSubmit, [onHandleMultiselectSubmit]);
+
+    useEffect(() => {
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            getChannels();
+        } else if (!loading && !term) {
```

Comment:
This effect only runs in the first render, where loading is always false and the term is always the empty string. This check doesn't seem to make much sense.